src/components/CreateQuestionView.tsx

```diff
@@ -0,0 +1,85 @@
+import { useAtom } from "jotai"
+import { useDisclosure } from "@mantine/hooks"
+import { Box, Flex, Group, Modal } from "@mantine/core"
+import { InteractiveQuestion } from "@metabase/embedding-sdk-react"
+
+import { ThemedButton } from "./ThemedButton"
+import { isSaveModalOpenAtom } from "../store/save"
+import { useState } from "react"
+
+export const CreateQuestionView = () => {
+  const [isVisualization, { toggle }] = useDisclosure(false)
+  const [isSaveModalOpen, setSaveModalOpen] = useAtom(isSaveModalOpenAtom)
+
+  const [isVisualizationButtonShown, setVisualizationButtonShown] =
+    useState(false)
+
+  const onApplyEditorChanges = () => {
+    toggle()
+
+    // Do not show the "show visualization/editor" button until the user has visualized the question.
+    // This prevents users from seeing the "Question not found" indicator.
+    if (!isVisualization) {
+      setVisualizationButtonShown(true)
+    }
```

Comment:
This prevents the user for accidentally seeing "Question not found", or an empty screen because visualization is not ready yet.

PS. I think we should not show "Question not found" in the SDK if `questionId` is null or undefined. This logic would be nevertheless needed to avoid a blank screen.